packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -214,14 +215,34 @@ export async function prepareContext(config: Configuration, env: { [key: string]
     context[cxapi.DISABLE_ASSET_STAGING_CONTEXT] = true;
   }
 
-  const bundlingStacks = config.settings.get(['bundlingStacks']) ?? ['**'];
+  const bundlingStacks = setBundlingStacks(config);
   context[cxapi.BUNDLING_STACKS] = bundlingStacks;
 
   debug('context:', context);
 
   return context;
 }
 
+function setBundlingStacks(config: Configuration) {
```

Comment:
This works but is hard to parse for a human. In general I prefer this style:

```ts
// explain how this is even possible
if (!config.command) {
  return [];
}

const stackCommands = [...];

// command doesn't operate on a stack selection
if (!stackCommands.includes(config.command) {
  return [];
}

// If we deploy, diff, synth or watch a list of stacks exclusively we skip
// bundling for all other stacks.
if (config.settings.get(['exclusively']) && config.settings.get(['STACKS'])) {
  return config.settings.get(['STACKS']);
}

return ['**'];
```

Reply:
agreed. i had just copied over the code from elsewhere. will update